src/TM/Engine.java

```diff
@@ -0,0 +1,122 @@
+package TM;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Engine {
+  private List<Instruction> instructionList = new ArrayList<>();
+  private List<Transaction> transactionList = new ArrayList<>();
+
+  public static void main(String[] args) throws IOException {
+    Engine taskManager = new Engine();
+    taskManager.initial("C:\\Users\\ztian\\Downloads\\ADB-project\\sample\\test1.txt");
+  }
+
+  private void initial(String inputFile) throws IOException {
+    readFile(inputFile, false);
+    parseInstructions(true);
+  }
+
+  private void readFile(String inputFile, boolean verbose) throws IOException{
+    FileInputStream fis = new FileInputStream(inputFile);
+    BufferedReader br = new BufferedReader(new InputStreamReader(fis));
+    String line;
+    int[] pos = new int[1];
+    while ((line = br.readLine()) != null) {
```

Comment:
We should use .trim() to remove whitespace firstly?
https://stackoverflow.com/questions/9222762/java-readline-whitespace